spacy_loggers/wandb.py

```diff
@@ -19,102 +23,112 @@ def wandb_logger_v4(
     run_name: Optional[str] = None,
     log_best_dir: Optional[str] = None,
     log_latest_dir: Optional[str] = None,
+    log_custom_stats: Optional[List[str]] = None,
 ):
-    try:
-        import wandb
+    wandb = _import_wandb()
 
-        # test that these are available
-        from wandb import init, log, join  # noqa: F401
-    except ImportError:
-        raise ImportError(
-            "The 'wandb' library could not be found - did you install it? "
-            "Alternatively, specify the 'ConsoleLogger' in the "
-            "'training.logger' config section, instead of the 'WandbLogger'."
+    def setup_logger(
+        nlp: "Language", stdout: IO = sys.stdout, stderr: IO = sys.stderr
+    ) -> Tuple[Callable[[Dict[str, Any]], None], Callable[[], None]]:
+        match_stat = matcher_for_regex_patterns(log_custom_stats)
+        run = _setup_wandb(
+            wandb,
+            nlp,
+            project_name,
+            remove_config_values=remove_config_values,
+            entity=entity,
         )
+        if run_name:
+            wandb.run.name = run_name
+
+        if log_dataset_dir:
+            _log_dir_artifact(
+                wandb, path=log_dataset_dir, name="dataset", type="dataset"
+            )
+
```

Comment:
Not as such, although we could do that in the future if we wanted to. The main reasoning was to reduce the side-effects of the helper functions - since we conditionally import the logging framework (as opposed to a top-level, global import), it make sense to pass them around as a module. We do a similar thing in Thinc and other libraries where we need to optionally support on 3rd party packages.